src/stp/Tactic.cpp

```diff
@@ -30,17 +30,16 @@ Status Tactic::update(StpInfo const &info) noexcept {
         return Status::Failure;
     }
 
+    // Update skill info
+    auto skill_info = calculateInfoForSkill(info);
+
     // the tactic will not be reset if it's the first skill; it will be reset as well if the
     // state machine is finished
-    if(skills.finished() || (skills.current_num() != 0 && shouldTacticReset(info))){
-        RTT_INFO("State Machine reset for current tactic for ID = ", info.getRobot()->get()->getId())
-        // TODO: messy reset, do it in the state machine
-        skills.skip_n(- skills.current_num());
+    if(skills.finished() || (skills.current_num() != 0 && shouldTacticReset(skill_info))){
+        RTT_INFO("State Machine reset for current tactic for ID = ", skill_info.getRobot()->get()->getId())
+        skills.reset();
```

Comment:
Considering the other requests, this could then change into *see suggestion*. There is no difference, but this makes it a bit cleaner (probably?)

```suggestion
        this.reset();
```